pkgs/applications/networking/iroh/default.nix

```diff
@@ -0,0 +1,78 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, stdenv
+, cmake
+, protobuf
+, llvmPackages
+, rocksdb
+, Security
+, DiskArbitration
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "iroh";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "n0-computer";
+    repo = "iroh";
+    rev = "v${version}";
+    sha256 = "sha256-hw9OTV+IE4TE3Ubtpw6HiUmapBYAKN33xgVA1siCgqM=";
+  };
+
+  cargoPatches = [
+    ./Cargo.toml.patch
+    ./Cargo.lock.patch
+  ];
+  cargoSha256 = "sha256-lTYv0oivEe892xu1v9nf8AzNUD7KQFkVGe61Hj7CtlQ=";
+
+  nativeBuildInputs = [
+    cmake
+    protobuf
+    llvmPackages.clang
+  ];
+  buildInputs = [
+  ] ++ lib.optionals stdenv.isDarwin [
+    Security
+    DiskArbitration
+  ];
+
+  LIBCLANG_PATH = "${llvmPackages.libclang.lib}/lib";
+
+  # link rocksdb dynamically?
+  ROCKSDB_INCLUDE_DIR = "${rocksdb}/include";
+  ROCKSDB_LIB_DIR = "${rocksdb}/lib";
+
+  postPatch = ''
```

Comment:
Can you write this as a path with placeholders, using substituteAll to replace the placeholders with `version` before applying a patch? `substituteInFile` is a bit brittle...